modules/firefox/hm.nix

```
@@ -159,12 +165,22 @@ in
     );
     warnings = eachTarget (
       { target, ... }:
+      let
+        missingProfiles = lib.lists.subtractLists (builtins.attrNames
+          config.programs.${target.path}.profiles
+        ) config.stylix.targets.${target.path}.profileNames;
+      in
       lib.optional
         (
           config.programs.${target.path}.enable
-          && config.stylix.targets.${target.path}.profileNames == [ ]
+          && missingProfiles != [ ]
+          && config.programs.${target.path}.enableProfileWarnings
```

For performance reasons (and because it seems easier to understand) it would be better to first guard with the general `enableProfileWarnings` option:

```suggestion
          && config.programs.${target.path}.enableProfileWarnings
          && missingProfiles != [ ]
```